pkg/detector/detector.go

```diff
@@ -320,6 +321,34 @@ func (d *ResourceDetector) OnAdd(obj interface{}) {
 
 // OnUpdate handles object update event and push the object to queue.
 func (d *ResourceDetector) OnUpdate(oldObj, newObj interface{}) {
+	unstructuredOldObj, err := helper.ToUnstructured(oldObj)
+	if err != nil {
+		klog.Errorf("Failed to transform oldObj, error: %v", err)
+		return
+	}
+
+	unstructuredNewObj, err := helper.ToUnstructured(newObj)
+	if err != nil {
+		klog.Errorf("Failed to transform newObj, error: %v", err)
+		return
+	}
+
+	oldBackup := unstructuredOldObj.DeepCopy()
+	newBackup := unstructuredNewObj.DeepCopy()
+
+	// Remove the status and some system defined mutable fields in metadata, including managedFields and resourceVersion.
+	// Refer to https://kubernetes.io/docs/reference/kubernetes-api/common-definitions/object-meta/#ObjectMeta for more details.
+	removedFields := [][]string{{"status"}, {"metadata", "managedFields"}, {"metadata", "resourceVersion"}}
+	for _, r := range removedFields {
+		unstructured.RemoveNestedField(oldBackup.Object, r...)
+		unstructured.RemoveNestedField(newBackup.Object, r...)
+	}
+
+	if reflect.DeepEqual(oldBackup, newBackup) {
+		klog.Infof("Object (kind=%s, %s/%s) no need to reconcile, intercept the event adding to worker queue", oldBackup.GetKind(), oldBackup.GetNamespace(), oldBackup.GetName())
```

Comment:
I suggest creating a new file(like `eventfilter.go`) and putting this logic into a new function, like
```golang
func SpecificationChanged(oldObj, newObj *unstructured.Unstructured) bool
```
After this, you don't need to mock anymore, as the function would be easier to test.

Reply:
Thanks for your advice, I've updated in my current commit